CycloneDDS build failure in ROS 2 CentOS builds

Starting on Sept 29 the ROS 2 builds on CentOS have been failing during the cyclonedds build.

[First example](https://ci.ros2.org/job/nightly_linux-centos_debug/548/) : [Most recent failure](https://ci.ros2.org/job/nightly_linux-centos_debug/556/)

The error that's breaking the build is this:

```
/home/jenkins-agent/workspace/nightly_linux-centos_debug/ws/src/eclipse-cyclonedds/cyclonedds/src/security/api/../../core/ddsi/include/dds/ddsi/ddsi_cfgelems.h:688:54: error: ‘struct ddsi_config_thread_properties_listelem’ has no member named ‘__sched_priority’; did you mean ‘sched_priority’?
     MEMBEROF(ddsi_config_thread_properties_listelem, sched_priority),
                                                      ^~~~~~~~~~~~~~
```

which thanks to some spelunking by @clalancette we've found is due to the fact that on CentOS 7, the `/usr/include/sched.h` file provided by the `glibc-headers-2.17-307.el7.1.x86_64` package includes the following on line 44:
```
#define sched_priority       __sched_priority
```

I've not dug in to figure out why this define exists or the precise chain of includes that brings it into ddsi_cfgelems.h. One workaround would be to rename the local `sched_priority` variable to something else like `schedule_priority`.

--- a/src/core/ddsi/include/dds/ddsi/ddsi_cfgelems.h
+++ b/src/core/ddsi/include/dds/ddsi/ddsi_cfgelems.h
@@ -685,7 +685,7 @@ static struct cfgelem thread_properties_sched_cfgelems[] = {
       "able to assign some of the privileged scheduling classes.</p>"),
     VALUES("realtime","timeshare","default")),
   STRING("Priority", NULL, 1, "default",
-    MEMBEROF(ddsi_config_thread_properties_listelem, sched_priority),
+    MEMBEROF(ddsi_config_thread_properties_listelem, schedule_priority),
     FUNCTIONS(0, uf_maybe_int32, 0, pf_maybe_int32),
     DESCRIPTION(
       "<p>This element specifies the thread priority (decimal integer or "

--- a/src/core/ddsi/include/dds/ddsi/ddsi_config.h
+++ b/src/core/ddsi/include/dds/ddsi/ddsi_config.h
@@ -121,7 +121,7 @@ struct ddsi_config_thread_properties_listelem {
   struct ddsi_config_thread_properties_listelem *next;
   char *name;
   ddsrt_sched_t sched_class;
-  struct ddsi_config_maybe_int32 sched_priority;
+  struct ddsi_config_maybe_int32 schedule_priority;
   struct ddsi_config_maybe_uint32 stack_size;
 };
 

--- a/src/core/ddsi/src/q_thread.c
+++ b/src/core/ddsi/src/q_thread.c
@@ -291,8 +291,8 @@ static dds_return_t create_thread_int (struct thread_state1 **ts1_out, const str
   ddsrt_threadattr_init (&tattr);
   if (tprops != NULL)
   {
-    if (!tprops->sched_priority.isdefault)
-      tattr.schedPriority = tprops->sched_priority.value;
+    if (!tprops->schedule_priority.isdefault)
+      tattr.schedPriority = tprops->schedule_priority.value;
     tattr.schedClass = tprops->sched_class; /* explicit default value in the enum */
     if (!tprops->stack_size.isdefault)
       tattr.stackSize = tprops->stack_size.value;

--- a/src/tools/ddsperf/ddsperf.c
+++ b/src/tools/ddsperf/ddsperf.c
@@ -587,18 +587,27 @@ static uint32_t pubthread (void *varg)
 
   baggage = init_sample (&data, 0);
   ihs = malloc (nkeyvals * sizeof (dds_instance_handle_t));
-  for (unsigned k = 0; k < nkeyvals; k++)
+  if (!register_instances)
   {
-    data.seq_keyval.keyval = (int32_t) k;
-    if (register_instances)
-      dds_register_instance (wr_data, &ihs[k], &data);
-    else
+    for (unsigned k = 0; k < nkeyvals; k++)
       ihs[k] = 0;
   }
+  else
+  {
+    for (unsigned k = 0; k < nkeyvals; k++)
+    {
+      data.seq_keyval.keyval = (int32_t) k;
+      if ((result = dds_register_instance (wr_data, &ihs[k], &data)) != DDS_RETCODE_OK)
+      {
+        printf ("dds_register_instance failed: %d\n", result);
+        fflush (stdout);
+        exit (2);
+      }
+    }
+  }
+
   data.seq_keyval.keyval = 0;
-
   tfirst = dds_time();
-
   uint32_t bi = 0;
   while (!ddsrt_atomic_ld32 (&termflag))
   {

--- a/src/tools/pubsub/pubsub.c
+++ b/src/tools/pubsub/pubsub.c
@@ -1069,7 +1069,7 @@ static void pub_do_auto(const struct writerspec *spec) {
     for (k = 0; (uint32_t) k < nkeyvals; k++) {
         d.seq_keyval.keyval = k;
         if(spec->register_instances) {
-            dds_register_instance(spec->wr, &handle[k], &d);
+            (void) dds_register_instance(spec->wr, &handle[k], &d);
         }
     }
 

